distributed/diagnostics/plugin.py

```diff
@@ -358,20 +361,106 @@ class PackageInstall(WorkerPlugin, abc.ABC):
     PipInstall
     """
 
-    INSTALLER: ClassVar[str]
+    WORKER_PLUGIN_CLASS: ClassVar[type]
 
+    installer: _PackageInstaller
+    name: str
+    restart_workers: bool
+
+    def __init__(
+        self,
+        installer: _PackageInstaller,
+        restart_workers: bool,
+    ):
+        self.installer = installer
+        self.restart_workers = restart_workers
+        self.name = f"{self.installer.INSTALLER}-install-{uuid.uuid4()}"
+
+    async def start(self, scheduler: Scheduler) -> None:
+        if not hasattr(scheduler, "_package_install_lock"):
+            scheduler._package_install_lock = asyncio.Lock()  # type: ignore[attr-defined]
```

Comment:
nit: instead of registering an attribute, this could also be a `_PackageInstaller` class attribute (i.e. a global thing)